src/clj/game/cards/agendas.clj

```diff
@@ -430,7 +430,10 @@
                                                      " from the grip"
                                                      " to the bottom of the stack."
                                                      " The Runner draws 1 card"))
-                         (draw state :runner eid 1)))
+                         (queue-event state :runner-hand-changed?)
+                         (wait-for
+                           (checkpoint state side)
```

Comment:
Is the checkpoint here tied to a timing structure?

Reply:
Yeah, there's a checkpoint where there's a comma in brain rewiring. It's more obvious in the rules text than the printed text.

` The Runner adds X cards from the grip at random to the bottom of the stack, then draws 1 card.`